### ISSUE TITLE
exclude_from_weight_decay for AdamW

### ISSUE THREAD

**Issue by markub3327** [keras-team/keras#16201]
Hello, 

  please can you add an `exclude_from_weight_decay` parameter to AdamW? Some parameters during weight decay must be excluded.

For more details please look [here](https://github.com/tensorflow/addons/issues/2681).

Thanks a lot.
Have a nice day.

**Comment by chunduriv** [comment]
@markub3327,

Can you please elaborate about your feature and please specify the use cases for this feature. Thanks!

**Comment by markub3327** [comment]
I need to exclude some parameters from the weight decay process. Useful during training the Transformer model. List of regex patterns of variables excluded from weight decay. Variables whose names contain a substring matching the pattern will be excluded.

[Here](https://github.com/microsoft/unilm/blob/master/beit/semantic_segmentation/backbone/beit.py#L415) is the use case.

The code behind exclude_from_weight_decay from tensorflow/addons is [here](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/weight_decay_optimizers.py#L245).

**Comment by jvishnuvardhan** [comment]
@markub3327 What is the difference between this issue when compared to https://github.com/keras-team/keras/issues/16195?

Can you please state the differences clearly or close this issue this is a duplicate of https://github.com/keras-team/keras/issues/16195. Thanks!

**Comment by bhack** [comment]
This is a proposal to add `exclude_from_weight_decay` in AdamW that was added in Keras 30 days ago (it was already in TFA).

https://github.com/keras-team/keras/issues/16195 is related to port  RectifiedAdam (a new optimizer) with `exclude_from_weight_decay`  in Keras from TFA

**Comment by markub3327** [comment]
@bhack  

Thanks a lot ... Very good explanation!

**Comment by haifeng-jin** [comment]
@chenmoneygithub Any comments on this?

**Comment by chenmoneygithub** [comment]
@markub3327 Thanks for reporting the issue! Are you willing to open a PR to add `exclude_from_weight_decay` to AdamW? Thanks!

**Comment by markub3327** [comment]
Yes. Thanks.
